pow-migration/src/monitor/mod.rs

```diff
@@ -24,19 +25,20 @@ pub const READY_PERCENTAGE: u8 = 80;
 ///   Sender: Validator address
 ///   Recipient: Burn address
 ///   Value: 1 Luna
-///   Data: TBD
+///   Data: Hash of the generated `GenesisConfig`
 ///
-pub fn generate_ready_tx(validator: String) -> OutgoingTransaction {
+pub fn generate_ready_tx(validator: String, hash: &Blake2bHash) -> OutgoingTransaction {
     info!(
         validator_address = validator,
+        pos_genesis_hash = hash.to_hex(),
         "Generating ready transaction"
     );
     OutgoingTransaction {
         from: validator,
         to: Address::burn_address().to_user_friendly_address(),
         value: 1, //Lunas
         fee: 0,
-        data: None,
+        data: Some(hash.to_hex()),
```

Comment:
Which is the reason why we want this in `hex` rather than the raw hash?

Reply:
The PoW RPC specs expects `OutgoingTransaction.data` to be a `String`. A raw `Blake2bHash` doesn't implement that. I could have gone for `to_string` but `to_hex` is something we control and have defined which is a little more consistent.